tutorials/pages/shimmerevm-hardhat.md

```diff
@@ -0,0 +1,208 @@
+```
+
+:::note
+
+The sample project used here comes from the [<ins>Hardhat Quickstart guide</ins>](https://hardhat.org/getting-started/#quick-start), as well as its instructions.
+
+:::
+
+You should see the following prompt:
+
+![Hardhat Setup](../static/shimmer-hardhat-setup.png 'Hardhat Setup')
+
+Choose the JavaScript project and go through these steps to compile, test and deploy the sample contract.
+
+:::note Typescript
+If you're comfortable with typescript, feel free to select it. There's not much difference for this project. However, some syntax might differ.
+:::
+
+Your hardhat project should look like something this:
+
+```md
+.
+├── README.md
+├── contracts
+│ └── Lock.sol
+├── hardhat.config.js
+├── package-lock.json
+├── package.json
+├── scripts
+│ └── deploy.js
+└── test
+└── Lock.js
+```
+
+---
+
+## Check the Contract
+
+The `contracts` folder contains `Lock.sol`, which is a sample contract which consistis of a simple digital lock, where users could only withdraw funds after a given period of time.
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// Uncomment this line to use console.log
+// import "hardhat/console.sol";
+
+contract Lock {
+    uint public unlockTime;
+    address payable public owner;
+
+    event Withdrawal(uint amount, uint when);
+
+    constructor(uint _unlockTime) payable {
+        require(
+            block.timestamp < _unlockTime,
+            "Unlock time should be in the future"
+        );
+
+        unlockTime = _unlockTime;
+        owner = payable(msg.sender);
+    }
+
+    function withdraw() public {
+        // Uncomment this line, and the import of "hardhat/console.sol", to print a log in your terminal
+        // console.log("Unlock time is %o and block timestamp is %o", unlockTime, block.timestamp);
+
+        require(block.timestamp >= unlockTime, "You can't withdraw yet");
+        require(msg.sender == owner, "You aren't the owner");
+
+        emit Withdrawal(address(this).balance, block.timestamp);
+
+        owner.transfer(address(this).balance);
+    }
+}
+```
+
+The above is an auto-generated smart contract by hardhat. You can replace it with your own smart contract. If you want to understand what the smart contract does, it's a Lock contract, which will lock funds for a specific time period and allows withdrawals only past that point.
+Some important things to note are data types. `event` means that we can subscribe to it like a webhook and wait for a ping to the client side whenever a withdrawal is triggered with `emit` from inside the function `withdraw()`.
+
+## Set Up the Contract
+
+1. Go to `hardhat.config.js`
+2. Update the `hardhat-config` with shimmer network details as shown below:
+
+```js
+module.exports = {
+  solidity: '0.8.18',
+  defaultNetwork: 'shimmerevm-testnet',
+  networks: {
+    'shimmerevm-testnet': {
+      url: 'https://json-rpc.evm.testnet.shimmer.network',
+      chainId: 1071,
+      accounts: [priv_key],
```

Comment:
I guess it would make sense to link or describe how to get a private key. Many IOTA devs may never have worked with SCs

Reply:
Maybe a separate post on how to extract your private key from wallet?

Reply:
Or for now just link to some guide on how to get it from a metamask account? 🤔

Reply:
Yeah that's what I meant. No need for our own guide right now 👍

Reply:
added